From 4454f463175dd4062e43f1ccb7b32ef9f131f845 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 22 Jan 2018 16:44:45 +0800
Subject: [PATCH 3297/5242] MLK-17473-3 drm/imx: dcss: remove
 'dcss_plane_mod_supported()'

commit  63c4ed6d7b931c3c8692a7ca0907a97db54dd808 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'dcss_plane_mod_supported()' function is duplicated with
another function 'dcss_plane_format_mod_supported()'. So remove
it and use 'dcss_plane_format_mod_supported()' to replace its
calling.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |   16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index c019f12..ae45217 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -170,16 +170,6 @@ static bool dcss_plane_format_mod_supported(struct drm_plane *plane,
 	.format_mod_supported = dcss_plane_format_mod_supported,
 };
 
-static bool dcss_plane_mod_supported(int type, uint64_t mod)
-{
-	if (type == DRM_PLANE_TYPE_OVERLAY)
-		return mod == DRM_FORMAT_MOD_VSI_G1_TILED ||
-		       mod == DRM_FORMAT_MOD_VSI_G2_TILED ||
-		       mod == DRM_FORMAT_MOD_VSI_G2_TILED_COMPRESSED;
-
-	return false;
-}
-
 static int dcss_plane_atomic_check(struct drm_plane *plane,
 				   struct drm_plane_state *state)
 {
@@ -218,8 +208,10 @@ static int dcss_plane_atomic_check(struct drm_plane *plane,
 	}
 
 	if ((fb->flags & DRM_MODE_FB_MODIFIERS) &&
-	    !dcss_plane_mod_supported(plane->type, fb->modifier)) {
-		DRM_DEBUG_KMS("Invalid modifier: %llx", fb->modifier);
+	    !plane->funcs->format_mod_supported(plane,
+				fb->format->format,
+				fb->modifier)) {
+		DRM_INFO("Invalid modifier: %llx", fb->modifier);
 		return -EINVAL;
 	}
 
-- 
1.7.9.5

